### DISCUSSION TITLE
Jasmine não invoca $service.$parent

### DISCUSSION THREAD

**Discussion by andreypatricio** [frontendbr/forum#712] [Perguntas e Respostas]
Eu tenho o seguinte cenário:

Uma classe de teste cujo objetivo é validar minha diretiva AngularJS...

```
describe('Directive Spec', function() {

    var $compile, $rootScope, directives;

    beforeEach(module('ui-first', 'ui-first.directive'));

    beforeEach(module(function(_$provide_) {
        _$provide_.value('$commons');
    }));

    beforeEach(inject(function(_$compile_, _$rootScope_) {
        $compile = _$compile_;
        $rootScope = _$rootScope_;
        directive = $compile("<ui-pop-dir></ui-pop-dir>")($rootScope);
        $rootScope.$digest();
    }));

    it('blablabla', function() {...});
});
```

_ui-first_ é meu modulo e _ui-first.directive_ minha declaração de template no arquivo _karma.conf.js_.

Na minha diretiva, tenho tenho um _$inject_ para o meu controller...

`Controller.$inject = ['$scope', '$commons'];`

Dentro do meu _$commons_, tenho um outro $injector...

`var promisseT = $injector.get('promisseT');`

Ao executar o teste, falha com o seguinte log:

> TypeError: undefined is not an object (evaluating '$commons.$promisseT') (line 34)
> Controller@C:/Users/user.user/git/hub/src/pop.directive.js:34:33

Ideias? :(

**Comment by felipefialho** [comment]
@woliveiras Você manja de Angular1?

**Comment by woliveiras** [comment]
Trabalhei com NG1, mas não mexia com testes, @LFeh 

:cry:

**Comment by andreypatricio** [comment]
@woliveiras e @LFeh - Consegui evoluir no problema. A princípio, estava fazendo o $inject incorreto. Ao constatar e injetar o serviço correto, avancei, mas me deparei com um novo impedimento. 

Não estou conseguindo mockar valores. Ao tentar reproduzir o valor do meu injetado, me deparo com o seguinte problema:

Ao utilizar:

`spyOnProperty(modalInstance.params, 'objeto').and.returnValues({})`

Obtenho...

> Error: spyOn could not find an object to spy upon for objeto (line 4555)

Esse meu _objeto_ está no meu controller, utilizado da seguinte forma:

` vm.objeto = angular.copy($modalInstance.params.objeto) || {};`

Continuamos na luta... :\

**Comment by safirelauene** [comment]
Olá @andreypatricio , tudo bem?
Você não está conseguindo mockar nenhum objeto ?
Ah, uma dúvida, você está usando Jasmine por aí?
Será que você consegue colocar aqui pra gente qual function você está querendo testar, acho que ajudaria entender melhor o cenário. :)
